server/audio_device.hpp

```diff
@@ -24,15 +24,17 @@ struct AudioDevice
 	{
 	}
 
-	AudioDevice(std::string s, std::string n, int i)
+	AudioDevice(std::string s, std::string n, std::string a, int i)
```

Comment:
Depending on what style you prefer, an [initialization list](http://en.cppreference.com/w/cpp/language/initializer_list) might be better here.